### DISCUSSION TITLE
mangle configuration does not work when zipping on different vms.

### DISCUSSION THREAD

**Discussion by com3345** [webpack/webpack#16870] [General]
Hi, all

I am using a CI pipeline which enables multiple vms as a compile pool to compile js projects.

As we are forced to remove binary differences on different compilation environments, so I turned on the configuration:

`mangle: ['exports']` to prevent my variables from changing when compressed.

But the outcome shows part of variables change if I compile my project on different vms in the pool.

For example:

`...q(n)?{unit:"minute",value:n}:(n=function(e,t){return(t.valueOf()-e.valueOf())/1e3}(e,t)...`

and

`...L(n)?{unit:"minute",value:n}:(n=function(e,t){return(t.valueOf()-e.valueOf())/1e3}(e,t)...`

Any help?

**Comment by alexander-akait** [comment]
There are two solutions:
- disable `mangle` for terser https://github.com/terser/terser#minify-options
- disable parallels for `terser-webpack-plugin`, warmup magle cache (one run to create cache) - take look at `nameCache` for `terser` and pass that cache to `terser`, but every time when you will add more things for mangle you need to regenerate cache
